snapcraft/internal/states/_pull_state.py

```diff
@@ -43,11 +43,13 @@ def _schema_properties():
 class PullState(State):
     yaml_tag = u'!PullState'
 
-    def __init__(self, property_names, part_properties=None, project=None):
+    def __init__(self, property_names, part_properties=None, project=None,
+                 pkg_list=None):
         # Save this off before calling super() since we'll need it
         # FIXME: for 3.x the name `schema_properties` is leaking
         #        implementation details from a higher layer.
         self.schema_properties = property_names
+        self.pkg_list = pkg_list
```

Comment:
Meh... I hate adding more properties to the state. Remember we're also planning on tracking VCS info here. Would it be better to nest all this asset tracking under a dict called `assets`? Then it would take a form like:

```python
assets = {
    'stage-packages': [<the package list you already have>],
    'source': <VCS stuff we have yet to do>,
}
```

That gives us room to expand to tracking more assets without further cluttering up the root of the state (since essentially `vars(state)` is written out as YAML). Thoughts?